File: app/src/main/java/com/byd/dashcast/cluster/mirror/InputDisplayRoutingPolicy.java
package com.byd.dashcast.cluster.mirror;

/** Safety decision for injecting an event only after its destination was explicitly applied. */
public final class InputDisplayRoutingPolicy {

    private InputDisplayRoutingPolicy() {}

    public static boolean canInject(int displayId, boolean targetApplied) {
        return displayId > 0 && targetApplied;
    }
}

File: app/src/main/java/com/byd/dashcast/cluster/mirror/InputDisplayRoutingPolicy.kt
package com.byd.dashcast.cluster.mirror

/** Safety decision for injecting an event only after its destination was explicitly applied. */
object InputDisplayRoutingPolicy {

    @JvmStatic
    fun canInject(displayId: Int, targetApplied: Boolean): Boolean {
        return displayId > 0 && targetApplied
    }
}
